src/renderers/dom/shared/eventPlugins/ChangeEventPlugin.js

```diff
@@ -289,6 +289,18 @@ function getTargetInstForInputOrChangeEvent(
   }
 }
 
+function handleControlledInputBlur(inst, node) {
+  // Fiber and ReactDOM keep wrapper state in separate places
+  let state = inst._wrapperState || node._wrapperState;
+
+  if (!state || !state.controlled || node.type !== 'number') {
+    return;
+  }
+
+  // If controlled, assign the value attribute to the current value on blur
+  node.setAttribute('value', '' + node.value);
```

Comment:
Won't this cause us to set the `value` attribute every time the input is blurred, even if the value hasn't changed?